src/ActionModal/components/DelegationModal.vue

```diff
@@ -189,38 +188,36 @@ export default {
     transactionData() {
       if (!this.from) return {}
 
-      if (this.from === this.modalContext.userAddress) {
+      if (this.from === this.session.address) {
         return {
           type: transaction.DELEGATE,
-          validatorAddress: this.validator.operator_address,
+          validatorAddress: this.validator.operatorAddress,
           amount: uatoms(this.amount),
-          denom: this.denom
+          denom: toMicroDenom(this.denom)
```

Comment:
Is this the correct approach?

Reply:
Sounds good so far

Reply:
should we let the API handle this instead?

Reply:
No, as this is the value going to the blockchain and it is signed client side.